gradle-jdks-setup/src/main/resources/gradle-jdks-setup.sh

```
@@ -84,7 +84,7 @@ read_value() {
   if [ ! -f "$1" ]; then
     die "ERROR: $1 not found, aborting Gradle JDK setup"
   fi
-  read -r value < "$1"
+  read -r value < "$1" || die "ERROR: Unable to read value from $1. Make sure the file ends with a newline."
```

reads the configuration files (eg. `gradle/jdks/17/macos/aarch64/download-url`)
`read` expects to have an endline at the end, otherwise it will return a non-zero exit code and because we have `set -o pipefail` enabled (see line#42), the script would fail silently